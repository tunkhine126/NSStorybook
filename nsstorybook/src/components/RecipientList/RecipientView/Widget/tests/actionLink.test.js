/**
 *
 * Tests for ActionLink
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import TestingComponent from 'components/shared/Testing';
import { render } from '@testing-library/react';
// import 'jest-dom/extend-expect'; // add some helpful assertions
import ActionLink from '../actionLink';

describe('<ActionLink />', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(
      <TestingComponent>
        <ActionLink message="addRecipientChild" />
      </TestingComponent>
    );
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <TestingComponent>
        <ActionLink message="addRecipientChild" />
      </TestingComponent>
    );
    expect(firstChild).toMatchSnapshot();
  });
});
