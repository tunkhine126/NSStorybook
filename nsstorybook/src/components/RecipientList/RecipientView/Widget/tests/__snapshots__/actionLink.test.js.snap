// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ActionLink /> Should render and match the snapshot 1`] = `
<button
  class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-addBtn-105 MuiButton-textPrimary makeStyles-textPrimary-162"
  tabindex="0"
  type="button"
>
  <span
    class="MuiButton-label makeStyles-label-161"
  >
    <span>
      Add Recipient
    </span>
  </span>
  <span
    class="MuiTouchRipple-root"
  />
</button>
`;
